src/client/component/auth.cpp

```diff
@@ -63,6 +63,7 @@ namespace auth
 		std::string get_key_entropy()
 		{
 			std::string entropy{};
+			entropy.append(std::to_string(static_cast<int>(utils::cryptography::random::get_integer())));
```

Comment:
Fair point, I hadn't thought of that. Is there a downside to having non-persistent GUIDs?

Alternatively, I could configure this so that the GUID is only calculated once, then stored/cached in the h1-mod Appdata folder, and accessed when needed. That way, the GUIDs are unique in the case of identical hardware combinations, but still mostly persistent.

Reply:
players need to be identified so that they can be banned from servers, or to track other stuff, if they aren't persistent then this is impossible

Reply:
Would the solution I proposed be acceptable? It's not perfect in that a player can delete the file to create a new GUID, but similarly with the current implementation, a player could relatively easily spoof hardware parameters to circumvent a ban, or containerize the application.

Changing IP is usually more time consuming or costly than spoofing hardware parameters. Would banning by IP be a preferable approach?

Obviously, no solution will be perfect, and even in a more traditional setting, users could make a new account upon being banned, but just to find a balance between a non-inhibiting and functional approach for user identification.

Reply:
most players dont even know how to compile the client, i doubt most of them would even attempt to spoof hardware parameters, but deleting a file is quite simple and would make it easier for players to circumvent bans imo. i dont think any changes to the guid system are necessary atm and if they are it should be on a separate pr (in general PRs should only address 1 issue at a time). 

the fix to the same ip issue is nice tho so if you could reduce this pr to just that it would be great

Reply:
Hardware spoofers are a fairly common approach to circumventing bans nowadays, especially with commercial anticheat software attempting to identify cheaters across multiple accounts.

https://github.com/topics/hwid-spoofer

Would you agree that banning by IP would be a more reliable approach? This is the approach taken by Call of Duty 4 servers, and by many other games. Changing IP at a minimum takes the same amount of time as using a hardware spoofer, and in the common case of using a VPN, would cost money to do. In contrast, popular hardware spoofers are freely available, and only need to be installed and activated.

The two PRs were combined due to being very small changes.

Reply:
whether hardware spoofers are available or not isnt that important, it matters how far someone will go to circumvent a ban, deleting a file is easy, having to install additional software may also be simple but requires more time and effort and people might be driven away from it just because of that. 
if someone really wanted to they could circumvent any ban they wanted, whether hardware or ip address based or anything else. 

banning by ip address may be effective but it may result in false bans since ip addresses are often shared or changed. either way bans are not handled by the client itself to this isnt relevant.

the size of the changes doesnt really matter, the protocol for these projects is to have separate PRs for each issue you want to fix

Reply:
The prevalence of hardware spoofers is highly relevant - any implementation of user identification will have a means of circumvention. However, the current approach to user identification is inhibiting,  and does not provide security beyond less inhibiting approaches.

Possibility of assignment of banned IPs to otherwise innocent individuals is certainly a downside of banning by IP. Similarly, with the present implementation, a player could be banned according to their hardware combination, and another player who happens to have that hardware combination would then be banned. However, user identification by GUID provides additional, unintended limitations to users (as described in the PR), and identification by IP address does not. As such, is user identification by IP address not preferable?

Bundling of tiny, obvious PRs is common across organizations I've participated in. Sorry for any inconvenience this may have caused.